samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -111,27 +112,35 @@ public LocalApplicationRunner(SamzaApplication app, Config config) {
    * @param metadataStoreFactory the instance of {@link MetadataStoreFactory} to read and write to coordinator stream.
    */
   public LocalApplicationRunner(SamzaApplication app, Config config, MetadataStoreFactory metadataStoreFactory) {
-    this(ApplicationDescriptorUtil.getAppDescriptor(app, config), getCoordinationUtils(config), metadataStoreFactory);
+    this(app, config, getCoordinationUtils(config), metadataStoreFactory);
```

Comment:
In the old flow, `getCoordinationUtils` seemed to be called with the full config. In the new flow, `getCoordinationUtils` will get called with the initial config, not the full config. Do you need to load the config before calling `getCoordinationUtils`?

Reply:
Very good question, and nice catch. You are right, both getCoordinationUtils as well as getDefaultCoordinatorStreamStoreFactory need to be updated to load full config first if applicable.